### PR TITLE
Add `html-preview-link` feature

### LINKED ISSUE #1096: Add button to preview HTML files
very useful for .html files

### PR DESCRIPTION
**Pull request by kidonng**
Closes #1096 

# Test

https://github.com/CodingTrain/website/blob/4f90eedb9618257d9166241e92e51a7f3f00a08e/code_challenges/PerlinNoiseTerrain_p5.js/index.html

### CODE REVIEW

File: source/features/html-preview-link.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+import {groupSiblings} from '../libs/group-buttons';
+
+function init(): void {
+	const rawButton = select('#raw-url') as HTMLAnchorElement;
+	if (rawButton && /\.html?/.test(rawButton.pathname)) {
```
Comment (fregante) [inline source/features/html-preview-link.tsx]: I think you can move this to the feature’s `include`, by creating a `isHTMLSingleFile` test function (not in the current file, not `page-detect.ts`

File: source/features/html-preview-link.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		rawButton
+			.parentElement! // `BtnGroup`
+			.prepend(
+				<a
+					className="btn btn-sm BtnGroup-item"
+					href={`https://cdn.statically.io/gh${link.join('/')}`}>
+					Preview
+				</a>
+			);
+		groupSiblings(rawButton);
```
Comment (fregante) [inline source/features/html-preview-link.tsx]: This probably isn’t necessary in this feature. Maybe it’s not necessary even in the feature you copied it from, but I don’t remember.

File: source/features/html-preview-link.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+import {groupSiblings} from '../libs/group-buttons';
+
+function init(): void {
+	const rawButton = select('#raw-url') as HTMLAnchorElement;
```
Comment (fregante) [inline source/features/html-preview-link.tsx]: You can use `select<HTMLAnchorElement>(...)!;

Comment (kidonng) [reply source/features/html-preview-link.tsx]: 😅 `rawButton.pathname` is used below where `pathname` is from `HTMLAnchorElement`

File: source/features/html-preview-link.tsx (hunk 2 above)
Comment (kidonng) [reply source/features/html-preview-link.tsx]: You mean `groupSiblings`? It's from `copy-file` feature. Will check whether it is needed later 😊

File: source/features/html-preview-link.tsx (hunk 1 above)
Comment (kidonng) [reply source/features/html-preview-link.tsx]: > I’m gonna guess that if (rawbutton) isn’t necessary in that case.

Yeah I doubted that when I first read it but didn't check, come to think it now, there will always be a raw button for files.

Comment (kidonng) [reply source/features/html-preview-link.tsx]: > I think you can move the regex test to the feature’s include, by creating a isSingleHTMLFile test function (not in the current file, not page-detect.ts

😅 Honestly, I don't know where to put it then, so I put it into `utils.ts`.

File: source/features/html-preview-link.tsx (hunk 2 above)
Comment (kidonng) [reply source/features/html-preview-link.tsx]: You're right #2506

**Comment by fregante** [comment]
Thank you! Finally! That issue was opened almost 2 years ago

**Pull request merged**